app/models/job_proxy_dispatcher.rb

```diff
@@ -167,30 +167,28 @@ def start_job_on_proxy(job, proxy)
     assign_proxy_to_job(proxy, job)
     _log.info "Job #{job.attributes_log}"
     job_options = {:args => ["start"], :zone => MiqServer.my_zone}
-    job_options.merge!(:server_guid => proxy.guid, :role => "smartproxy") if proxy.kind_of?(MiqServer)
+    job_options[:server_guid => proxy.guid]
+    job_options[:role => "smartproxy"]
     @active_vm_scans_by_zone[MiqServer.my_zone] += 1
     queue_signal(job, job_options)
   end
 
   def assign_proxy_to_job(proxy, job)
     job.agent_id        = proxy.id
-    job.agent_class     = proxy.class.to_s
     job.agent_name      = proxy.name
     job.started_on      = Time.now.utc
     job.dispatch_status = "active"
     job.save
 
     # Increment the counts for busy proxies and busy hosts for embedded
-    busy_proxies["#{job.agent_class}_#{job.agent_id}"] ||= 0
-    busy_proxies["#{job.agent_class}_#{job.agent_id}"] += 1
+    busy_proxies["MiqServer_#{job.agent_id}"] ||= 0
+    busy_proxies["MiqServer_#{job.agent_id}"] += 1
 
     # Track the host/vc resource for embedded scans so we can limit the resource impact
-    if proxy.kind_of?(MiqServer)
-      key = embedded_scan_resource(@vm)
-      if key
-        busy_resources_for_embedded_scanning[key] ||= 0
-        busy_resources_for_embedded_scanning[key] += 1
-      end
+    key = embedded_scan_resource(@vm)
+    if key
```

Comment:
sometimes we do:

```ruby
if (key = embedded_scan_resource(@vm))
```

use if you like (and the other edits end up in the same commit)
(you're not really changing this line, so punting is good too)